specification/quota/resource-manager/Microsoft.Quota/preview/2021-03-15-preview/quota.json

```diff
@@ -0,0 +1,1226 @@
+          "type": "string"
+        },
+        "display": {
+          "$ref": "#/definitions/OperationDisplay"
+        },
+        "origin": {
+          "type": "string"
+        }
+      }
+    },
+    "OperationDisplay": {
+      "type": "object",
+      "properties": {
+        "provider": {
+          "type": "string",
+          "description": "Provider name."
+        },
+        "resource": {
+          "type": "string",
+          "description": "Resource name."
+        },
+        "operation": {
+          "type": "string",
+          "description": "Operation name."
+        },
+        "description": {
+          "type": "string",
+          "description": "Operation description."
+        }
+      }
+    },
+    "CommonResourceProperties": {
+      "type": "object",
+      "description": "Resource properties.",
+      "properties": {
+        "id": {
+          "description": "Resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource type. \"Microsoft.Quota/quotaLimits\""
+        }
+      }
+    },
+    "CurrentQuotaLimit": {
+      "type": "object",
+      "description": "Quota limit.",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "quotaInformation": {
+          "description": "Quota information details.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/CurrentQuotaLimitBase"
+        },
+        "properties": {
+          "description": "Additional properties for the quota request status for the specified resource.",
+          "$ref": "#/definitions/QuotaRequestStatusDetails",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "CurrentQuotaLimitBase": {
+      "type": "object",
+      "description": "Quota limit.",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "properties": {
+          "description": "Quota properties for the specified resource.",
+          "$ref": "#/definitions/QuotaProperties"
+        }
+      }
+    },
+    "QuotaProperties": {
+      "description": "Quota properties for the specified resource.",
+      "type": "object",
+      "properties": {
+        "limit": {
+          "description": "Quota limit.",
+          "type": "integer",
+          "format": "int32",
+          "readOnly": false
+        },
+        "currentValue": {
+          "description": "Usage information for the current resource.",
+          "type": "integer",
+          "format": "int32",
+          "readOnly": true
+        },
+        "unit": {
+          "description": " The quota limit units, such as Count and Bytes. When requesting quota, use the **unit** value returned in the GET response in the request body of your PUT operation.",
+          "type": "string",
```

Comment:
>string [](http://example.com/codeflow?start=19&length=6)

Should `unit` be modeled as an enum?

Reply:
We are planning to onboard all 40+ RPs. 
We don't know all the possible values  used currently and what new RP will use in future. That's why we have kept it as string.
Thus we can onboard new RP, with any swagger changes.

Reply:
Makes sense.